src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -949,4 +986,78 @@ mod tests {
             }
         }
     }
+
+    fn msrs_from_entries(msr_entries: &[(u32, u64)]) -> Msrs {
+        Msrs::from_entries(
+            &msr_entries
+                .iter()
+                .map(|&(index, data)| kvm_msr_entry {
+                    index,
+                    data,
+                    ..Default::default()
+                })
+                .collect::<Vec<_>>(),
+        )
+        .unwrap()
+    }
+
+    fn assert_msrs(msr_chunks: &[Msrs], expected_msr_entries: &[(u32, u64)]) {
+        let flattened_msrs = msr_chunks.iter().flat_map(|msrs| msrs.as_slice());
+        let expected_msrs: Vec<(u32, u64)> = expected_msr_entries.to_vec();
```

Comment:
You don't really need to copy here. Just used `expected_msr_entries` for iteration.

Reply:
true, done